pkgs/top-level/agda-packages.nix

```diff
@@ -13,7 +16,10 @@ let
 
     lib = lib.extend (final: prev: import ../build-support/agda/lib.nix { lib = prev; });
 
-    agda = withPackages [] // { inherit withPackages; };
+    agda = withPackages [] // {
+      inherit withPackages;
+      passthru.tests.all = withPackages (lib.filter (pkg: agdaPackages.lib.isUnbrokenAgdaPackage pkg) (lib.attrValues agdaPackages));
```

Comment:
This looks a little dodgy now that I look at it. I think `self` here should be `agdaPackages` (I could be totally wrong on this)

Reply:
Sorry, I can't follow. Where does `self` occur here?

Reply:
it's further up the file

Reply:
oh I mistyped, the `agdaPackages` can probably be replaced with `self` (other way round to what I said)

Reply:
But this file doesn't have `self`. Maybe you mean a different place?

Reply:
https://github.com/NixOS/nixpkgs/blob/7c1b01613fd205cdebe459e119b3b702891769c7/pkgs/top-level/agda-packages.nix#L8 here it is

Reply:
Alright, now I got it thanks!